cdap-app-fabric/src/main/java/io/cdap/cdap/app/store/Store.java

```diff
@@ -296,6 +297,15 @@ Map<ProgramRunId, RunRecordDetail> getRuns(ApplicationId applicationId, ProgramR
    */
   void addApplication(ApplicationId id, ApplicationSpecification specification);
 
+  /**
+   * Creates new application if it doesn't exist. Updates existing one otherwise.
+   *
+   * @param id            application id
+   * @param specification application specification to store
+   * @param owner         user name (owner) of the application version
+   */
+  void addApplication(ApplicationId id, ApplicationSpecification specification, String owner);
+
```

Comment:
new function definition to write owner information (fetched from authentication context) to the table (when the new owner column is added later).

Reply:
This should be unblocked? Since the enabler PR is merged, we could add needed columns now

Reply:
yes, this will be the next rev

Reply:
Can we have a default implementation of the other `addApplication` method to just call this one? This would make the DefaultStore implementation looks simpler. Or you can also just change the `addApplication` method to add owner without introducing a new method, as this is an internal interface only.

Reply:
I didn't want to change `addApplication` because it might break more tests (will need to be changed)- and since owner is only optional, I thought that is not needed. Instead I'll add a default implementation